client/src/main/java/io/pravega/client/tables/BadKeyVersionException.java

```diff
@@ -0,0 +1,25 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.client.tables;
+
+/**
+ * Exception that is thrown whenever a conditional {@link KeyValueTable} update failed due to the provided key version
+ * mismatching. This is different from {@link NoSuchKeyException}.
+ */
+public class BadKeyVersionException extends ConditionalTableUpdateException {
```

Comment:
`NoSuchKeyException` is another type of `ConditionalTableUpdateException` that indicates the Key does not exist when it was expected to exist. In some cases it is useful to the caller to know that the key does not exist instead of just "BadVersion". The Controller already uses this differentiation.

Reply:
Check out here (branch forked from this that implements the Table Segment Client) on how Wire commands are converted to such exceptions:
https://github.com/andreipaduroiu/pravega/blob/issue-4333-tables-segment-client/client/src/main/java/io/pravega/client/tables/impl/TableSegmentImpl.java#L227